### PR TITLE
Storage contract

### LINKED ISSUE #29: Storage Smart Contract

### PR DESCRIPTION
**Pull request by vlasp**

### CODE REVIEW

**Review by srawulwar — APPROVED** [review]
lgtm

**Comment by techster10173** [comment]
[Storage integration into Three0 core](https://trello.com/c/mR8Ny8LG/25-storage-integration-into-three0-core)

File: src/components/storage-components/FileManager.tsx [hunk 1]
```diff
@@ -21,40 +24,110 @@ const defaultLabels = {
...
+        metadata: fileMetadata,
+        path: filepath,
+        receiver_id: window.walletConnection.account().accountId,
+        //   perpetual_royalties: royalties
+      },
+      "300000000000000", // attached GAS (optional)
+      "100000000000000000000000" // attached deposit in yoctoNEAR (optional)
+    );
+
+    return new Promise((resolve) => {
```
Comment (techster10173) [inline src/components/storage-components/FileManager.tsx:91]: Return the contract call itself

File: src/components/storage-components/FileManager.tsx [hunk 2]
```diff
@@ -21,40 +24,110 @@ const defaultLabels = {
...
+      setTimeout(resolve, 100);
+    });
+  };
+
+  const rename = (path: string) =>
```
Comment (techster10173) [inline src/components/storage-components/FileManager.tsx:96]: Add TODO comment

File: src/components/storage-components/FileManager.tsx [hunk 3]
```diff
@@ -21,40 +24,110 @@ const defaultLabels = {
...
+    new Promise((resolve) => {
+      setTimeout(resolve, 100);
+    });
+
+  const deletePaths = (paths: string) =>
```
Comment (techster10173) [inline src/components/storage-components/FileManager.tsx:101]: Add TODO comment

**Review by techster10173 — APPROVED** [review]
LGTM!

**Pull request merged**